### ISSUE TITLE
coalesce in 'order by' results in not returning request

### ISSUE THREAD

**Issue by jukart** [crate/crate#7085]
**CrateDB version**: 2.3.3

**Problem description**:

Using coalesce with timestamps in order by results in a request which never returns.
In the crate log this excpetion occures:

**Traceback**:
```
java.lang.UnsupportedOperationException: Unsupported data type: timestamp
	at io.crate.execution.engine.sort.SortSymbolVisitor.missingObject(SortSymbolVisitor.java:243) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.sort.SortSymbolVisitor.access$300(SortSymbolVisitor.java:56) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.sort.SortSymbolVisitor$1.newComparator(SortSymbolVisitor.java:196) ~[crate-app-2.3.3.jar:2.3.3]
	at org.apache.lucene.search.SortField.getComparator(SortField.java:361) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.FieldValueHitQueue.<init>(FieldValueHitQueue.java:140) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.FieldValueHitQueue.<init>(FieldValueHitQueue.java:32) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.FieldValueHitQueue$OneComparatorFieldValueHitQueue.<init>(FieldValueHitQueue.java:62) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.FieldValueHitQueue.create(FieldValueHitQueue.java:163) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.TopFieldCollector.create(TopFieldCollector.java:376) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at org.apache.lucene.search.TopFieldCollector.create(TopFieldCollector.java:328) ~[lucene-core-6.6.1.jar:6.6.1 9aa465a89b64ff2dabe7b4d50c472de32c298683 - varunthacker - 2017-08-29 21:54:39]
	at io.crate.execution.engine.collect.collectors.LuceneOrderedDocCollector.initialSearch(LuceneOrderedDocCollector.java:132) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.collectors.LuceneOrderedDocCollector.collect(LuceneOrderedDocCollector.java:115) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.collectors.OrderedDocCollector.get(OrderedDocCollector.java:61) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.collectors.OrderedLuceneBatchIteratorFactory.loadFromAllUnExhausted(OrderedLuceneBatchIteratorFactory.java:161) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.collectors.OrderedLuceneBatchIteratorFactory.access$100(OrderedLuceneBatchIteratorFactory.java:51) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.collectors.OrderedLuceneBatchIteratorFactory$Factory.tryFetchMore(OrderedLuceneBatchIteratorFactory.java:101) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.distribution.merge.BatchPagingIterator.loadNextBatch(BatchPagingIterator.java:112) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.CloseAssertingBatchIterator.loadNextBatch(CloseAssertingBatchIterator.java:69) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.LimitingBatchIterator.loadNextBatch(LimitingBatchIterator.java:66) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.AsyncOperationBatchIterator.uncheckedLoadNextBatch(AsyncOperationBatchIterator.java:134) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.AsyncOperationBatchIterator.loadNextBatch(AsyncOperationBatchIterator.java:125) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.MappedForwardingBatchIterator.loadNextBatch(MappedForwardingBatchIterator.java:52) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.action.sql.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:79) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.action.sql.RowConsumerToResultReceiver.accept(RowConsumerToResultReceiver.java:47) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.InterceptingRowConsumer.tryForwardResult(InterceptingRowConsumer.java:81) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.InterceptingRowConsumer.accept(InterceptingRowConsumer.java:68) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.data.ListenableRowConsumer.accept(ListenableRowConsumer.java:40) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.pipeline.ProjectingRowConsumer.accept(ProjectingRowConsumer.java:105) ~[crate-app-2.3.3.jar:2.3.3]
	at io.crate.execution.engine.collect.BatchIteratorCollectorBridge$SyncBatchItCollector.doCollect(BatchIteratorCollectorBridge.java:64) ~[crate-app-2.3.3.jar:2.3.3]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) ~[crate-app-2.3.3.jar:2.3.3]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
```

**Steps to reproduce**:

```
CREATE TABLE test (
  t1 TIMESTAMP,
  t2 TIMESTAMP
)
SELECT * FROM test ORDER BY COALESCE(t1, t2);
```

**Comment by m-kharbat** [comment]
Thank you for reporting, we will look into this.

**Comment by matriv** [comment]
@jukart The issue has been fixed in ``2.3.5`` with https://github.com/crate/crate/commit/12328ad9d1ae6918a732bc4d13e5b9faa7537fb9
